argo/providers/stub/provider.go

```diff
@@ -0,0 +1,94 @@
+package stub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/sky-uk/kfp-operator/providers/base"
+	"github.com/sky-uk/kfp-operator/providers/base/generic"
+)
+
+type StubProviderConfig struct {
+	ExpectedOutput base.Output `yaml:"expectedOutput"`
+	ExpectedId     string      `yaml:"expectedId"`
```

Comment:
`ExpectedOutput` should be `StubbedOutput`
`ExpectedId` should move into a stuct like `ExpectedInput.Id`